crates/holochain/src/conductor/cell/validation_package.rs

```diff
@@ -36,14 +43,126 @@ impl ValidationPackageDb {
     }
 }
 
+#[instrument(skip(header_hashed, env, ribosome, conductor_api, network))]
 pub(super) async fn get_as_author(
-    header: Header,
+    header_hashed: HeaderHashed,
     env: EnvironmentRead,
-    dna_file: &DnaFile,
+    ribosome: &impl RibosomeT,
     conductor_api: &impl CellConductorApiT,
+    network: &HolochainP2pCell,
 ) -> CellResult<ValidationPackageResponse> {
+    let header = header_hashed.as_content();
+
     // Get the source chain with public data only
-    let source_chain = SourceChain::public_only(env)?;
+    let source_chain = SourceChain::public_only(env.clone())?;
+
+    // Get the header data
+    let (app_entry_type, header_seq) = match header
+        .entry_type()
+        .cloned()
+        .map(|et| (et, header.header_seq()))
+    {
+        Some((EntryType::App(aet), header_seq)) => (aet, header_seq),
+        _ => return Ok(None.into()),
+    };
+
+    //Get entry def
+    let entry_def = get_entry_def_from_ids(
+        app_entry_type.zome_id(),
+        app_entry_type.id(),
+        ribosome.dna_file(),
+        conductor_api,
+    )
+    .await?;
+
+    // Get the required validation package
+    let required_validation_type = match entry_def {
+        Some(ed) => ed.required_validation_type,
+        None => return Ok(None.into()),
+    };
+
+    // Gather the package
+    match required_validation_type {
+        RequiredValidationType::Element => {
+            // TODO: I'm not sure if we should handle this case, it seems like they should already have the element
+            Ok(None.into())
+        }
+        RequiredValidationType::SubChain => Ok(Some(get_as_author_sub_chain(
+            header_seq,
+            app_entry_type,
```

Comment:
Just a musing: I wonder if we don't want to just define the SubChain with a QueryFilter rather than an app entry type?

Reply:
We could do I guess